include/fmt/compile.h

```diff
@@ -31,6 +31,22 @@ class compiled_string {};
 template <typename S>
 struct is_compiled_string : std::is_base_of<compiled_string, S> {};
 
+#ifdef __cpp_if_constexpr
+# define FMT_COMPILE_IMPL(s) FMT_STRING_IMPL(s, fmt::detail::compiled_string)
+#else
+namespace fmt_compile_error {
+  template<class T> struct false_type : std::false_type {};
```

Comment:
Could it be done without introducing this additional type?

Reply:
Do you mean false_type? It's optional, but error messages of GCC and clang will be a bit uglier, because in case of static_assert, its condition is highlighted. Like following:
```
static_assert(!std::is_same<T,T>::value, "FMT_COMPILE requires C++17 `constexpr if` compiler support"); return ""; }
 |                   ^~~~~~~~~~~~~~~~~~~~
```
If code in the condition is not obvious then it attends too many unwanted attention. I fear decisions with storing value locally in the named constexpr bool or constant alias don't work on the all supported platforms. Maybe they do. I'm not sure.
```
  template<class T> FMT_CONSTEXPR const char* report_compile_error() {
    using false_v = !std::is_same<T,T>::value;
    // constexpr bool false_v = !std::is_same<T,T>::value;
    static_assert(
        false_v, "FMT_COMPILE requires C++17 `constexpr if` compiler support"
    );
    return "";
  }
```

Reply:
Btw, I added additional namespace fmt_compile_error just not to spoil namespace with the specific code.